### DISCUSSION TITLE
Re-structure env-variables to correct format for JSON

### DISCUSSION THREAD

**Discussion by Ashta0** [community/community#26854] [Actions]
<p>Hi all, I’m fairly new in coding. Anyway, I am trying to compare two branches and get the JIRA Tickets or JIRA Keys between those two branches whenever a pull request is made and I have already set as env-variable. However, whenever I pass it’s value to JSON, it isn’t correctly formatted. Is this possible? Please see the sample below.</p>
<pre><code class="lang-auto">#Git Command to query the Jira tickets.
 - name: Tickets
      run: echo ::set-env name=Tickets::$(git log origin/BranchA..origin/BranchB --pretty=oneline | perl -ne '{ /(\w+)-(\d+)/ &amp;&amp; print "$1-$2\n" }' | sort | uniq)

#Webhook to JIRA
 - name: Update tickets
      uses: sendwebhook
      with:
        url: ""
        json: '{"issues": "${{ env.Tickets }}", "and some field to update"}'
</code></pre>
<p>The value that was retrieved is like this:</p>
<p><code>json: {"issues": "JIRAKey-1 JIRAKey-2 JIRAKey-3" }</code></p>
<p>instead of like this:</p>
<p><code>json: {"issues": "JIRAKey-1", "JIRAKey-2", "JIRAKey-3" }</code></p>

**Comment by airtower-luna** [comment, marked as answer]
<p>When you use <code>${{ env.Tickets }}</code> what happens is a simple string substitution, the literal value of the <code>Tickets</code> environment variable is placed at the location. If you want it contain a JSON structure you’ll have to create that yourself. Note that to get a valid JSON array of issues your <code>json</code> field should look like this:</p>
<pre><code class="lang-json">{"issues": ["JIRAKey-1", "JIRAKey-2", "JIRAKey-3"]}
</code></pre>
<p>I like Python, so I’d use something like this:</p>
<pre><code class="lang-yaml">- name: Create ticket list
  shell: python3 {0}
  run: |
    import json
    import os
    print(f'::set-env name=TICKET_LIST::{json.dumps(os.environ["Tickets"].split())}')
</code></pre>
<p>Well, I’d probably integrate that with parsing the <code>git log</code> output, but I think the idea is visible.</p>

**Comment by Ashta0** [comment]
<p>Thanks! This works very well. I’m thinking of investing time on Python more as well. :slight_smile: Have a great day!</p>

**Comment by subscriptionm** [comment]
Jira is an incredibly powerful tool for developers and testers alike. It's easy to create tracking algorithms and code in it. The possibilities are endless regarding what you can do with Jira.

**Reply by subscriptionm** [reply]
There are so many plugins and integrations available that you can tailor to Jira to fit your needs. If you are searching for such plugins to integrate, you can search on [Datafloq](https://datafloq.com/read/leveraging-new-data-capabilities-tableau-connector-jira/).  And if you can't find what you're looking for, you can always create your own plugin or integration. This is truly a tool that can be molded to fit any workflow.
